src/Peachpie.CodeAnalysis/CodeGen/Symbols/MethodSymbol.cs

```diff
@@ -6,29 +6,58 @@
 using System.Text;
 using Microsoft.CodeAnalysis;
 using Pchp.CodeAnalysis.CodeGen;
+using Pchp.CodeAnalysis.Semantics;
 
 namespace Pchp.CodeAnalysis.Symbols
 {
     internal abstract partial class MethodSymbol
     {
         static TypeSymbol EmitCreateRoutine(CodeGenerator cg, MethodSymbol method)
         {
-            var overloads = ImmutableArray<MethodSymbol>.Empty;
+            var il = cg.Builder;
 
-            if (method is AmbiguousMethodSymbol a)
+            if (method.IsStatic)
             {
-                Debug.Assert(a.IsOverloadable);
-                method = a.Ambiguities[0];
-                overloads = a.Ambiguities.RemoveAt(0);
-            }
+                var overloads = ImmutableArray<MethodSymbol>.Empty;
+                if (method is AmbiguousMethodSymbol a)
+                {
+                    Debug.Assert(a.IsOverloadable);
+                    method = a.Ambiguities[0];
+                    overloads = a.Ambiguities.RemoveAt(0);
+                }
 
-            var il = cg.Builder;
+                // CreateUserRoutine("method", ldtoken method, new[] { ldtoken methodOverload1, ... })
+                il.EmitStringConstant(method.MetadataName);
+                cg.EmitLoadToken(method, null);
+                cg.Emit_NewArray(cg.CoreTypes.RuntimeMethodHandle, overloads, m => cg.EmitLoadToken(m, null));
 
-            il.EmitStringConstant(method.MetadataName);
-            cg.EmitLoadToken(method, null);
-            cg.Emit_NewArray(cg.CoreTypes.RuntimeMethodHandle, overloads, m => cg.EmitLoadToken(m, null));
+                return cg.EmitCall(ILOpCode.Call, cg.CoreMethods.Reflection.CreateUserRoutine_string_RuntimeMethodHandle_RuntimeMethodHandleArr);
+            }
+            else
+            {
+                ImmutableArray<MethodSymbol> overloads;
+                if (method is AmbiguousMethodSymbol a)
+                {
+                    Debug.Assert(a.IsOverloadable);
+                    overloads = a.Ambiguities;
+                }
+                else
+                {
+                    overloads = ImmutableArray.Create(method);
+                }
+
+                // CreateUserRoutine("method", new[] { (MethodInfo)MethodBase.GetMethodFromHandle(ldtoken method), ... })
+                il.EmitStringConstant(method.MetadataName);
+                cg.Emit_NewArray(cg.CoreTypes.MethodInfo, overloads, m =>
+                {
+                    cg.EmitLoadToken(m, null);
+                    cg.EmitCall(ILOpCode.Call, cg.CoreMethods.Reflection.GetMethodFromHandle_RuntimeMethodHandle);
+                    cg.EmitCastClass(cg.CoreTypes.MethodInfo);
```

Comment:
GetWellKnownType( ... ) instead

Reply:
Updated in 1c87acd2ce3b892f3abef5337cef31cf3451e891.